dataobj/schedule.cc

```diff
@@ -149,8 +151,13 @@ bool schedule_t::append(const grund_t* gr, uint8 minimum_loading, uint16 waiting
 		return false;
 	}
 	if(is_stop_allowed(gr)) {
-		entries.append(schedule_entry_t(gr->get_pos(), minimum_loading, waiting_time_shift, stop_flags), 4);
-		return true;
+		if( !next_line.is_bound() ) {
+			entries.append(schedule_entry_t(gr->get_pos(), minimum_loading, waiting_time_shift, stop_flags), 4);
+			return true;
+		} else {
+			entries.insert_at(entries.get_count()-1, schedule_entry_t(gr->get_pos(), minimum_loading, waiting_time_shift, stop_flags));
+			return true;
+		}
```

Comment:
(nit) ifの条件がtrue/false両方の場合を扱うときは、特に理由がなければ条件文は `!` で反転させないようにしましょう。
```suggestion
		if(  next_line.is_bound()  ) {
			entries.insert_at(entries.get_count()-1, schedule_entry_t(gr->get_pos(), minimum_loading, waiting_time_shift, stop_flags));
		} else {
			entries.append(schedule_entry_t(gr->get_pos(), minimum_loading, waiting_time_shift, stop_flags), 4);
		}
		return true;
```